sunpy/instr/lyra.py

```diff
@@ -1,123 +1,581 @@
+              numpy recarray
+            Artifacts which were found but not removed as they were not
+            included when user defined artifacts kwarg.
+        artifact_status["not_found"] = artifacts not found : list of strings
+            Artifacts listed to be removed by user when defining artifacts
+            kwarg which were not found in time series time range.
+
+    References
+    ----------
+    [1] http://proba2.oma.be/data/TARDIS
+
+    Examples
+    --------
+    Remove LARs (Large Angle Rotations) from LYRALightCurve for 4-Dec-2014.
+        >>> lc = sunpy.lightcurve.LYRALightCurve.create("2014-12-02")
+        >>> lc_nolars = remove_artifacts_from_lyralightcurve(lc, artifacts=["LAR"])
+
+    To also retrieve information on the artifacts during that day...
+        >>> lc_nolars, artifact_status = remove_artifacts_from_lyralightcurve(
+                lc, artifacts=["LAR"], return_artifacts=True)
+
+    """
+    # Check that input argument is of correct type
+    if not isinstance(lc, lightcurve.LightCurve):
+        raise TypeError("lc must be a LightCurve object.")
+    # Remove artifacts from time series
+    data_columns = lc.data.columns
+    time, channels, artifact_status = _remove_lytaf_events(
+        lc.data.index,
+        channels=[np.asanyarray(lc.data[col]) for col in data_columns],
+        artifacts=artifacts, return_artifacts=True, lytaf_path=lytaf_path,
+        force_use_local_lytaf=force_use_local_lytaf)
+    # Create new copy copy of lightcurve and replace data with
+    # artifact-free time series.
+    lc_new = copy.deepcopy(lc)
+    lc_new.data = pandas.DataFrame(
+        index=time, data=dict((col, channels[i])
+                              for i, col in enumerate(data_columns)))
+    if return_artifacts:
+        return lc_new, artifact_status
+    else:
+        return lc_new
+
+def _remove_lytaf_events(time, channels=None, artifacts=[],
+                         return_artifacts=False, fitsfile=None,
+                         csvfile=None, filecolumns=None,
+                         lytaf_path=LYTAF_PATH, force_use_local_lytaf=False):
+    """
+    Removes periods of LYRA artifacts from a time series.
+
+    This functions removes periods corresponding to certain artifacts recorded
+    in the LYRA annotation file from an array of times given by the time input.
+    If a list of arrays of other properties is supplied through the channels
+    kwarg, then the relevant values from these arrays are also removed.  This
+    is done by assuming that each element in each array supplied corresponds to
+    the time in the same index in time array.  The artifacts to be removed are
+    given via the artifacts kwarg.  The default is "all", meaning that all
+    artifacts will be removed.  However, a subset of artifacts can be removed
+    by supplying a list of strings of the desired artifact types.
+
+    Parameters
+    ----------
+    time : ndarray/array-like of datetime objects
```

Comment:
ndarray should be `<backtick>numpy.ndarray<backtick>` and datetime should also be in backticks.